toolchain/mfc/test/test.py

```
@@ -181,103 +211,230 @@ def test():
     if ARG("dry_run"):
         cons.print(f"  [bold magenta]{case.get_uuid()}[/bold magenta]     SKIP     {case.trace}")
+        timeout_timer.cancel()
         return
 
-    cmd = case.run([PRE_PROCESS, SIMULATION], gpus=devices)
-
-    out_filepath = os.path.join(case.get_dirpath(), "out_pre_sim.txt")
+    try:
+        # Check timeout before starting
+        if timeout_flag.is_set():
+            raise TestTimeoutError("Test case exceeded 1 hour timeout")
```

Ruff (0.14.7)

Abstract `raise` to an inner function

(TRY301)